module.go

```diff
@@ -136,6 +132,26 @@ func getServer(_, addr string) (*tsnetServerDestructor, error) {
 	return s.(*tsnetServerDestructor), nil
 }
 
+func getAuthKey(host string) string {
+	hostAuthKey, loaded := app.LoadOrStore(host, "")
```

Comment:
Any reason you wouldn't just pass the auth key in here? You have 2 LordOrStore operations which my intuition tells me breaks the atomicity...

Reply:
Since the global key is semi-static I should be able to fetch it outside this function and use it if the host specific key doesn't exist.

My assumption was the Caddy App setup happens semi-serially and that these would all be already loaded which would mean we should still get back the same values? Or am I misunderstanding the atomicity issue

Reply:
Oh, I see, so the `storedAuthKey` and `hostAuthKey` are different values? In that case, it's probably fine. Just make sure to call Delete() in your module's `Cleanup()` method so that it can free up unused memory.